pkg/controller/user-cluster-controller-manager/resources/reconciler.go

```diff
@@ -631,7 +633,11 @@ func (r *reconciler) reconcileDeployments(ctx context.Context) error {
 		return fmt.Errorf("failed to reconcile Deployments in namespace %s: %v", kubernetesdashboard.Namespace, err)
 	}
 
-	kubeSystemCreators := []reconciling.NamedDeploymentCreatorGetter{coredns.DeploymentCreator()}
+	kubeSystemCreators := []reconciling.NamedDeploymentCreatorGetter{
+		coredns.DeploymentCreator(),
+		envoyagent.DeploymentCreator(),
```

Comment:
We should deploy the agent conditionally, only when the cluster uses the tunneling expose strategy.